packages/flutter/test/services/set_method_call_handler_test.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // We need a separate test file for this test case (instead of including it
+  // in platform_channel_test.dart) since we rely on the WidgetsFlutterBinding
+  // not being initialized and we call ensureInitialized() in the other test
+  // file.
+  test('throws assertion error if WidgetsFlutterBinding is not yet initialized', () {
+    const MethodChannel methodChannel = MethodChannel('mock');
+
+    // This used to throw _CastError:<Null check operator used on a null value>
```

Comment:
It is weird to comment retrospectively, reader without knowledge of previous version of code, which is usually the case, may confused at this comment.

Maybe
```
Verify the exception is thrown when setMethodCallHandler is called before ServiceBinding is initialized.
```

Reply:
Sure, makes sense. I have a good formulation in mind 👍